### ISSUE TITLE
IntFlag iteration behaviour is inconsistent between versions

### ISSUE THREAD

**Issue by dignissimus** [python/cpython#93210]
On 3.10, iterating an IntFlag enum will include multi-bit aliases but on 3.11 and above, multi-bit aliases aren't included. Since IntFlag enums shouldn't list multi-bit aliases 3.10 should be patched so its behaviour matches that of 3.11 and 3.12.

```python
sam@samtop ~/Documents/git/cpython (git)-[v3.11.0a5~234] % ./python                                             
Python 3.11.0a4+ (tags/v3.11.0a4-30-g83d544b929:83d544b929, May 21 2022, 09:41:53) [GCC 12.1.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from enum import IntFlag
>>> class E(IntFlag):
...     x = 3
... 
>>> list(E)
[]
>>> 
```

```python
Python 3.10.4 (main, Mar 23 2022, 23:05:40) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from enum import IntFlag
>>> class E(IntFlag):
...     x = 3
... 
>>> list(E)
[<E.x: 3>]
>>> 
```

Tested on Linux with 3.10 and compared with behaviour in 3.11 and 3.12

Copied from #93035

---------------------------

https://github.com/python/cpython/commit/83d544b9292870eb44f6fca37df0aa351c4ef83a introduced the error when creating the enum but the underlying issue is that multi-bit flags aren't included when iterating the enum. eea8148b7dff5ffc7b84433859ac819b1d92a74d Introduced support for multi-bit members but they weren't listed with the enum. On 3.10 this was fixed in 9bf7c2d638a582af2444bc864feba13ab8957b68 and remained fixed in 2a9ab75af32b1ee9f210ae2a0718990687d0f79d. But it was never fixed in 3.11 and above.

```python
sam@samtop ~/Documents/git/cpython (git)-[v3.11.0a5~234] % ./python                                             
Python 3.11.0a4+ (tags/v3.11.0a4-30-g83d544b929:83d544b929, May 21 2022, 09:41:53) [GCC 12.1.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from enum import IntFlag
>>> class E(IntFlag):
...     x = 3
... 
>>> list(E)
[]
>>> 
```

```python
Python 3.10.4 (main, Mar 23 2022, 23:05:40) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from enum import IntFlag
>>> class E(IntFlag):
...     x = 3
... 
>>> list(E)
[<E.x: 3>]
>>> 
```
The problem might be around here
https://github.com/python/cpython/blob/b96e20c1d9be4e6d5ea3e48c9c97e5ecd02f6055/Lib/enum.py#L713-L727

I will try looking more into it later and I think an extra test should be added for this.

_Originally posted by @dignissimus in https://github.com/python/cpython/issues/93035#issuecomment-1133591254_
**Bug report**

--------------------

Actually, @ethanfurman, which is (or should be?) the correct behaviour when listing IntFlag enums with multi-bit members? This is what happens on Python 3.10.4
```python
sam@samtop ~ % python          
Python 3.10.4 (main, Mar 23 2022, 23:05:40) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from enum import IntFlag
>>> class E(IntFlag):
...     x = 1
...     y = 3
... 
>>> list(E)
[<E.x: 1>, <E.y: 3>]
>>> 
```
And on trunk
```python
sam@samtop ~/Documents/git/cpython (git)-[main] % ./python
Python 3.12.0a0 (heads/main:fa2b8b75eb, May 21 2022, 10:35:37) [GCC 12.1.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from enum import IntFlag
>>> class E(IntFlag):
...     x = 1
...     y = 3
... 
>>> list(E)
[<E.x: 1>]
```

_Originally posted by @dignissimus in https://github.com/python/cpython/issues/93035#issuecomment-1133593022_

-----------------------------------

@ethanfurman How should the following work? (should it work?)
```python
def test_bizarre(self):   
    class Bizarre(Flag):
        b = 3
        c = 4
        d = 6
    self.assertEqual(repr(Bizarre(7)), '<Bizarre.d|c|b: 7>')
```
My current patch produces the following
```python
>>> repr(Bizarre(7))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/sam/Documents/git/cpython/Lib/enum.py", line 396, in __call__
    return cls.__new__(cls, value)
  File "/home/sam/Documents/git/cpython/Lib/enum.py", line 729, in __new__
    raise exc
  File "/home/sam/Documents/git/cpython/Lib/enum.py", line 711, in __new__
    result = cls._missing_(value)
  File "/home/sam/Documents/git/cpython/Lib/enum.py", line 857, in _missing_
    possible_member = cls._create_pseudo_member_(value)
  File "/home/sam/Documents/git/cpython/Lib/enum.py", line 872, in _create_pseudo_member_
    raise ValueError("%r is not a valid %s" % (value, cls.__qualname__))
ValueError: 7 is not a valid Bizarre
```
Behaviour with single-bit members is correct
```python
>>> class Bizarre(Flag):
...     x = 1
...     y = 2
...     z = 4
... 
>>> Bizarre(7)
<Bizarre.z|y|x: 7>
>>> 
```

_Originally posted by @dignissimus in https://github.com/python/cpython/issues/93035#issuecomment-1135198643_

-------------------

**Comment by ambv** [comment]
Unless @ethanfurman decides otherwise, I don't think we should be changing this behavior in 3.10.6 as the first six releases of 3.10 have different behavior. You see, it would be tricky for library authors to work around this behavior if it changed in a bug fix release. It's because some Linux distributions will backport selected bug fixes from, say, 3.10.5, while still reporting being an older bug fix Python release (say, 3.10.3).

We need @ethanfurman to tell us which behavior is correct. If he says that it's the 3.11+ one, it's still cleaner to admit defeat with 3.10 and document the difference in behavior stating that in 3.11+ it's different.

cc 3.10 Release Manager @pablogsal

**Comment by ethanfurman** [comment]
Multi-bit aliases not showing up is a bug-fix for 3.11+, but because 3.10 is so far in it's life cycle we will not be fixing it there.
